packages/compiler/src/core/types.ts

```diff
@@ -280,7 +280,7 @@ export type NeverIndexer = {
 };
 
 export type ModelIndexer = {
-  readonly key: Scalar;
+  readonly key: Enum | Scalar | Union;
```

Comment:
does this mean the key here could be an enum, or scalar or union of enums and/or scalars?
How should we understand the `Union` here?
would a union of models be OK here?
Also I did not see any test cases in this PR when key is an enum or union, therefore I am really curious what the `union` here stands for

Reply:
so the union here actually means union of literals of enums and scalars.
if the compiler could prevent all those unexpected usage scenarios, this is actually fine

Reply:
Indeed. The compiler can check the key type for `Map` types, but IIRC the compiler currently doesn't check argument types for `@indexer` decorator as it's an intrinstics decorator without any declarations.